### ISSUE TITLE
Problem with emerging 'dev-haskell/text-icu-0.7.0.1': 'error: ‘true’ undeclared (first use in this function)'

### ISSUE THREAD

**Issue by hereGittyGitty** [gentoo-haskell/gentoo-haskell#1119]
Hi there... I am having trouble with emerging/updating `dev-haskell/text-icu-0.7.0.1` since a couple of days. Besides some _deprecated warnings_ which become errors there is also the one output from the title which I believe is the root of the failure. On the _IRC_ channel _slyfox_ also pointed me to this [commit](https://github.com/gentoo-haskell/gentoo-haskell/commit/12f31c988574501cf434ad9dd480dea19dd51bc8#commitcomment-45452804.). 

Here is some info about my system:

* `emerge --info '=dev-haskell/text-icu-0.7.0.1::haskell'`

```
Portage 3.0.9 (python 3.8.6-final-0, default/linux/amd64/17.1/desktop/gnome/systemd, gcc-9.3.0, glibc-2.32-r3, 5.4.80-gentoo-r1-x86_64 x86_64)
=================================================================
                         System Settings
=================================================================
System uname: Linux-5.4.80-gentoo-r1-x86_64-x86_64-Intel-R-_Core-TM-_i7-8565U_CPU_@_1.80GHz-with-glibc2.2.5
KiB Mem:    32687636 total,  23423352 free
KiB Swap:   49151996 total,  49151996 free
Timestamp of repository gentoo: Mon, 28 Dec 2020 12:00:01 +0000
Head commit of repository gentoo: ede9b9f147f30b4d174ca3bfb2d7eb944faeee1a
Timestamp of repository brother-overlay: Mon, 28 Dec 2020 10:28:07 +0000
Head commit of repository brother-overlay: 6ebf8d0beac4aace75cd58f60dfe5f9c633add3a

Timestamp of repository haskell: Mon, 28 Dec 2020 10:27:27 +0000
Head commit of repository haskell: 15c7e2e3355749a21880d060fb2900b4de9d741b

Head commit of repository lolo-overlay: f1a4a02c64f3eef3b18b977be03de41eb8b20f72

sh bash 5.0_p18
ld GNU ld (Gentoo 2.34 p6) 2.34.0
app-shells/bash:          5.0_p18::gentoo
dev-java/java-config:     2.3.1::gentoo
dev-lang/perl:            5.30.3::gentoo
dev-lang/python:          2.7.18-r4::gentoo, 3.8.6::gentoo, 3.9.0::gentoo
dev-util/cmake:           3.17.4-r1::gentoo
sys-apps/baselayout:      2.7::gentoo
sys-apps/sandbox:         2.20::gentoo
sys-devel/autoconf:       2.13-r1::gentoo, 2.69-r5::gentoo
sys-devel/automake:       1.16.2-r1::gentoo
sys-devel/binutils:       2.34-r2::gentoo
sys-devel/gcc:            9.3.0-r2::gentoo
sys-devel/gcc-config:     2.3.2-r1::gentoo
sys-devel/libtool:        2.4.6-r6::gentoo
sys-devel/make:           4.3::gentoo
sys-kernel/linux-headers: 5.4-r1::gentoo (virtual/os-headers)
sys-libs/glibc:           2.32-r3::gentoo
Repositories:

gentoo
    location: /var/db/repos/gentoo
    sync-type: rsync
    sync-uri: rsync://rsync.gentoo.org/gentoo-portage
    priority: -1000
    sync-rsync-verify-metamanifest: yes
    sync-rsync-verify-jobs: 1
    sync-rsync-extra-opts: 
    sync-rsync-verify-max-age: 24

brother-overlay
    location: /var/db/repos/brother-overlay
    sync-type: git
    sync-uri: https://github.com/gentoo-mirror/brother-overlay.git
    masters: gentoo

haskell
    location: /var/db/repos/haskell
    sync-type: git
    sync-uri: https://github.com/gentoo-mirror/haskell.git
    masters: gentoo

lolo-overlay
    location: /var/db/repos/lolo-overlay
    sync-type: git
    sync-uri: https://github.com/hereGittyGitty/lolo-overlay.git
    masters: gentoo
    priority: 50

Installed sets: @my_qt5-addons, @my_qt5-essentials, @my_qt5-tools
ACCEPT_KEYWORDS="amd64"
ACCEPT_LICENSE="@FREE CC-Sampling-Plus-1.0"
CBUILD="x86_64-pc-linux-gnu"
CFLAGS="-march=skylake -mtune=skylake -O2 -pipe -frecord-gcc-switches"
CHOST="x86_64-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/lib64/libreoffice/program/sofficerc /usr/share/gnupg/qualified.txt"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/php/apache2-php7.4/ext-active/ /etc/php/cgi-php7.4/ext-active/ /etc/php/cli-php7.4/ext-active/ /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo /etc/texmf/language.dat.d /etc/texmf/language.def.d /etc/texmf/updmap.d /etc/texmf/web2c"
CXXFLAGS="-march=skylake -mtune=skylake -O2 -pipe -frecord-gcc-switches"
DISTDIR="/var/cache/distfiles"
ENV_UNSET="CARGO_HOME DBUS_SESSION_BUS_ADDRESS DISPLAY GOBIN GOPATH PERL5LIB PERL5OPT PERLPREFIX PERL_CORE PERL_MB_OPT PERL_MM_OPT XAUTHORITY XDG_CACHE_HOME XDG_CONFIG_HOME XDG_DATA_HOME XDG_RUNTIME_DIR"
FCFLAGS="-march=skylake -mtune=skylake -O2 -pipe -frecord-gcc-switches"
FEATURES="assume-digests binpkg-docompress binpkg-dostrip binpkg-logs buildpkg config-protect-if-modified distlocks ebuild-locks fixlafiles ipc-sandbox merge-sync multilib-strict network-sandbox news parallel-fetch pid-sandbox preserve-libs protect-owned qa-unresolved-soname-deps sandbox sfperms split-elog strict unknown-features-warn unmerge-logs unmerge-orphans userfetch userpriv usersandbox usersync xattr"
FFLAGS="-march=skylake -mtune=skylake -O2 -pipe -frecord-gcc-switches"
GENTOO_MIRRORS="https://mirror.eu.oneandone.net/linux/distributions/gentoo/gentoo/ https://mirror.netcologne.de/gentoo/ https://ftp.fau.de/gentoo"
LANG="en_US.UTF-8"
LDFLAGS="-Wl,-O1 -Wl,--as-needed"
PKGDIR="/var/cache/binpkgs"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages --exclude=/.git"
PORTAGE_TMPDIR="/var/tmp"
USE="X a52 aac acl acpi alsa amd64 bash-completion berkdb bluetooth branding bzip2 cairo cdda cdr cli colord crypt cups dbus dri dts dvd dvdr eds emboss encode evo exif flac fortran gdbm gif gnome gnome-keyring gnome-online-accounts gpm gstreamer gtk gui hscolour iconv icu introspection ipv6 jpeg lcms libglvnd libnotify libsecret libtirpc mad mng mp3 mp4 mpeg multilib nautilus ncurses networkmanager nls nptl offensive ogg opengl openmp pam pango pcre pdf png policykit ppds pulseaudio qt5 readline sdl seccomp spell split-usr ssl startup-notification svg sysprof systemd tcpd tiff truetype udev udisks unicode upower usb vim-syntax vorbis wayland wxwidgets x264 xattr xcb xml xv xvid zlib zsh-completion" ABI_X86="64" ADA_TARGET="gnat_2018" ALSA_CARDS="ali5451 als4000 atiixp atiixp-modem bt87x ca0106 cmipci emu10k1x ens1370 ens1371 es1938 es1968 fm801 hda-intel intel8x0 intel8x0m maestro3 trident usb-audio via82xx via82xx-modem ymfpci" APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_alias authn_anon authn_dbm authn_default authn_file authz_dbm authz_default authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir disk_cache env expires ext_filter file_cache filter headers include info log_config logio mem_cache mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias" CALLIGRA_FEATURES="karbon sheets words" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="aes avx avx2 f16c fma3 mmx mmxext pclmul popcnt sse sse2 sse3 sse4_1 sse4_2 ssse3" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock greis isync itrax mtk3301 nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf skytraq superstar2 timing tsip tripmate tnt ublox ubx" GRUB_PLATFORMS="efi-64 pc" INPUT_DEVICES="libinput" KERNEL="linux" L10N="de en fr" LCD_DEVICES="bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text" LIBREOFFICE_EXTENSIONS="presenter-console presenter-minimizer" LUA_SINGLE_TARGET="lua5-1" LUA_TARGETS="lua5-1" OFFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php7-2 php7-3 php7-4" POSTGRES_TARGETS="postgres10 postgres11" PYTHON_SINGLE_TARGET="python3_8" PYTHON_TARGETS="python2_7 python3_8" RUBY_TARGETS="ruby25 ruby26" USERLAND="GNU" VIDEO_CARDS="intel i965" XTABLES_ADDONS="quota2 psd pknock lscan length2 ipv4options ipset ipp2p iface geoip fuzzy condition tee tarpit sysrq steal rawnat logmark ipmark dhcpmac delude chaos account"
Unset:  CC, CPPFLAGS, CTARGET, CXX, EMERGE_DEFAULT_OPTS, INSTALL_MASK, LC_ALL, LINGUAS, MAKEOPTS, PORTAGE_BINHOST, PORTAGE_BUNZIP2_COMMAND, PORTAGE_COMPRESS, PORTAGE_COMPRESS_FLAGS, PORTAGE_RSYNC_EXTRA_OPTS
```

- `emerge -pqv '=dev-haskell/text-icu-0.7.0.1::haskell'`

```
[ebuild  N    ] dev-haskell/text-icu-0.7.0.1  USE="hscolour -doc -hoogle -profile" 
```

- output of complete build log

```
 * Package:    dev-haskell/text-icu-0.7.0.1
 * Repository: haskell
 * Maintainer: haskell@gentoo.org
 * USE:        abi_x86_64 amd64 elibc_glibc hscolour kernel_linux userland_GNU
 * FEATURES:   network-sandbox preserve-libs sandbox userpriv usersandbox
>>> Unpacking source...
>>> Unpacking text-icu-0.7.0.1.tar.gz to /var/tmp/portage/dev-haskell/text-icu-0.7.0.1/work
>>> Source unpacked in /var/tmp/portage/dev-haskell/text-icu-0.7.0.1/work
>>> Preparing source in /var/tmp/portage/dev-haskell/text-icu-0.7.0.1/work/text-icu-0.7.0.1 ...
 * Applying text-icu-0.7.0.1-icu.patch ...
 [ ok ]
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-haskell/text-icu-0.7.0.1/work/text-icu-0.7.0.1 ...
 * Using cabal-3.2.1.0.
 * Prepending /usr/lib64/ghc-8.10.3 to LD_LIBRARY_PATH
/usr/bin/ghc -package Cabal-3.2.1.0 --make /var/tmp/portage/dev-haskell/text-icu-0.7.0.1/work/text-icu-0.7.0.1/Setup.lhs -j9 +RTS -A256M -qb0 -RTS -threaded -dynamic -o setup
[1 of 1] Compiling Main             ( /var/tmp/portage/dev-haskell/text-icu-0.7.0.1/work/text-icu-0.7.0.1/Setup.lhs, /var/tmp/portage/dev-haskell/text-icu-0.7.0.1/work/text-icu-0.7.0.1/Setup.o )
Linking setup ...
./setup configure --ghc --prefix=/usr --with-compiler=/usr/bin/ghc --with-hc-pkg=/usr/bin/ghc-pkg --prefix=/usr --libdir=/usr/lib64 --libsubdir=text-icu-0.7.0.1/ghc-8.10.3 --datadir=/usr/share/ --datasubdir=text-icu-0.7.0.1/ghc-8.10.3 --ghc-options=-j9 +RTS -A256M -qb0 -RTS --with-ar=x86_64-pc-linux-gnu-ar --ghc-option=-optc-march=skylake --ghc-option=-optc-mtune=skylake --ghc-option=-optc-O2 --ghc-option=-optc-pipe --ghc-option=-optc-frecord-gcc-switches --ghc-option=-optl-Wl,-O1 --ghc-option=-optl-Wl,--as-needed --disable-executable-stripping --docdir=/usr/share/doc/text-icu-0.7.0.1 --verbose --enable-shared --enable-executable-dynamic --sysconfdir=/etc --disable-library-stripping
Using Parsec parser
Configuring text-icu-0.7.0.1...
Dependency base ==4.*: using base-4.14.1.0
Dependency bytestring -any: using bytestring-0.10.12.0
Dependency deepseq -any: using deepseq-1.4.4.0
Dependency text >=0.9.1.0: using text-1.2.4.1
Source component graph: component lib
Configured component graph:
    component text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm
        include base-4.14.1.0
        include bytestring-0.10.12.0
        include deepseq-1.4.4.0
        include text-1.2.4.1
Linked component graph:
    unit text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm
        include base-4.14.1.0
        include bytestring-0.10.12.0
        include deepseq-1.4.4.0
        include text-1.2.4.1
        Data.Text.ICU=text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm:Data.Text.ICU,Data.Text.ICU.Break=text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm:Data.Text.ICU.Break,Data.Text.ICU.Char=text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm:Data.Text.ICU.Char,Data.Text.ICU.Collate=text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm:Data.Text.ICU.Collate,Data.Text.ICU.Convert=text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm:Data.Text.ICU.Convert,Data.Text.ICU.Error=text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm:Data.Text.ICU.Error,Data.Text.ICU.Normalize=text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm:Data.Text.ICU.Normalize,Data.Text.ICU.Regex=text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm:Data.Text.ICU.Regex,Data.Text.ICU.Types=text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm:Data.Text.ICU.Types
Ready component graph:
    definite text-icu-0.7.0.1-DIipBkytmEBLlGes9uBsFm
        depends base-4.14.1.0
        depends bytestring-0.10.12.0
        depends deepseq-1.4.4.0
        depends text-1.2.4.1
Using Cabal-3.2.1.0 compiled by ghc-8.10
Using compiler: ghc-8.10.3
Using install prefix: /usr
Executables installed in: /usr/bin
Libraries installed in: /usr/lib64/text-icu-0.7.0.1/ghc-8.10.3
Dynamic Libraries installed in: /usr/lib64/x86_64-linux-ghc-8.10.3
Private executables installed in:
/usr/libexec/x86_64-linux-ghc-8.10.3/text-icu-0.7.0.1
Data files installed in: /usr/share/text-icu-0.7.0.1/ghc-8.10.3
Documentation installed in: /usr/share/doc/text-icu-0.7.0.1
Configuration files installed in: /etc
Using alex version 3.2.5 found on system at: /usr/bin/alex
Using ar given by user at: /usr/bin/x86_64-pc-linux-gnu-ar
No c2hs found
No cpphs found
No doctest found
Using gcc version 9.3.0 found on system at: /usr/bin/x86_64-pc-linux-gnu-gcc
Using ghc version 8.10.3 given by user at: /usr/bin/ghc
Using ghc-pkg version 8.10.3 given by user at: /usr/bin/ghc-pkg
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.24.0 found on system at: /usr/bin/haddock
Using happy version 1.20.0 found on system at: /usr/bin/happy
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.68 found on system at: /usr/bin/hpc
Using hsc2hs version 0.68.7 found on system at: /usr/bin/hsc2hs
Using hscolour version 1.24 found on system at: /usr/bin/HsColour
No jhc found
Using ld found on system at: /usr/bin/x86_64-pc-linux-gnu-ld
Using pkg-config version 1.6.3 found on system at: /usr/bin/pkg-config
Using runghc version 8.10.3 found on system at: /usr/bin/runghc
Using strip version 2.34 found on system at: /usr/bin/strip
Using tar found on system at: /bin/tar
No uhc found
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-haskell/text-icu-0.7.0.1/work/text-icu-0.7.0.1 ...
./setup build
Preprocessing library for text-icu-0.7.0.1..
Building library for text-icu-0.7.0.1..
[ 1 of 21] Compiling Data.Text.ICU.Break.Types ( Data/Text/ICU/Break/Types.hs, dist/build/Data/Text/ICU/Break/Types.o, dist/build/Data/Text/ICU/Break/Types.dyn_o )
[ 2 of 21] Compiling Data.Text.ICU.Collate.Internal ( Data/Text/ICU/Collate/Internal.hs, dist/build/Data/Text/ICU/Collate/Internal.o, dist/build/Data/Text/ICU/Collate/Internal.dyn_o )
[ 3 of 21] Compiling Data.Text.ICU.Error.Internal ( dist/build/Data/Text/ICU/Error/Internal.hs, dist/build/Data/Text/ICU/Error/Internal.o, dist/build/Data/Text/ICU/Error/Internal.dyn_o )
[ 4 of 21] Compiling Data.Text.ICU.Error ( dist/build/Data/Text/ICU/Error.hs, dist/build/Data/Text/ICU/Error.o, dist/build/Data/Text/ICU/Error.dyn_o )
[ 5 of 21] Compiling Data.Text.ICU.Convert.Internal ( Data/Text/ICU/Convert/Internal.hs, dist/build/Data/Text/ICU/Convert/Internal.o, dist/build/Data/Text/ICU/Convert/Internal.dyn_o )
[ 6 of 21] Compiling Data.Text.ICU.Internal ( dist/build/Data/Text/ICU/Internal.hs, dist/build/Data/Text/ICU/Internal.o, dist/build/Data/Text/ICU/Internal.dyn_o )
[ 7 of 21] Compiling Data.Text.ICU.Convert ( Data/Text/ICU/Convert.hs, dist/build/Data/Text/ICU/Convert.o, dist/build/Data/Text/ICU/Convert.dyn_o )
[ 8 of 21] Compiling Data.Text.ICU.Collate ( dist/build/Data/Text/ICU/Collate.hs, dist/build/Data/Text/ICU/Collate.o, dist/build/Data/Text/ICU/Collate.dyn_o )
[ 9 of 21] Compiling Data.Text.ICU.Collate.Pure ( Data/Text/ICU/Collate/Pure.hs, dist/build/Data/Text/ICU/Collate/Pure.o, dist/build/Data/Text/ICU/Collate/Pure.dyn_o )
[10 of 21] Compiling Data.Text.ICU.Break ( dist/build/Data/Text/ICU/Break.hs, dist/build/Data/Text/ICU/Break.o, dist/build/Data/Text/ICU/Break.dyn_o )
[11 of 21] Compiling Data.Text.ICU.Break.Pure ( Data/Text/ICU/Break/Pure.hs, dist/build/Data/Text/ICU/Break/Pure.o, dist/build/Data/Text/ICU/Break/Pure.dyn_o )
[12 of 21] Compiling Data.Text.ICU.Iterator ( Data/Text/ICU/Iterator.hs, dist/build/Data/Text/ICU/Iterator.o, dist/build/Data/Text/ICU/Iterator.dyn_o )
[13 of 21] Compiling Data.Text.ICU.Normalize.Internal ( dist/build/Data/Text/ICU/Normalize/Internal.hs, dist/build/Data/Text/ICU/Normalize/Internal.o, dist/build/Data/Text/ICU/Normalize/Internal.dyn_o )
[14 of 21] Compiling Data.Text.ICU.Normalize ( dist/build/Data/Text/ICU/Normalize.hs, dist/build/Data/Text/ICU/Normalize.o, dist/build/Data/Text/ICU/Normalize.dyn_o )
[15 of 21] Compiling Data.Text.ICU.Char ( dist/build/Data/Text/ICU/Char.hs, dist/build/Data/Text/ICU/Char.o, dist/build/Data/Text/ICU/Char.dyn_o )
[16 of 21] Compiling Data.Text.ICU.Regex.Internal ( dist/build/Data/Text/ICU/Regex/Internal.hs, dist/build/Data/Text/ICU/Regex/Internal.o, dist/build/Data/Text/ICU/Regex/Internal.dyn_o )
[17 of 21] Compiling Data.Text.ICU.Regex ( Data/Text/ICU/Regex.hs, dist/build/Data/Text/ICU/Regex.o, dist/build/Data/Text/ICU/Regex.dyn_o )
[18 of 21] Compiling Data.Text.ICU.Regex.Pure ( Data/Text/ICU/Regex/Pure.hs, dist/build/Data/Text/ICU/Regex/Pure.o, dist/build/Data/Text/ICU/Regex/Pure.dyn_o )
[19 of 21] Compiling Data.Text.ICU.Text ( Data/Text/ICU/Text.hs, dist/build/Data/Text/ICU/Text.o, dist/build/Data/Text/ICU/Text.dyn_o )
[20 of 21] Compiling Data.Text.ICU    ( Data/Text/ICU.hs, dist/build/Data/Text/ICU.o, dist/build/Data/Text/ICU.dyn_o )
[21 of 21] Compiling Data.Text.ICU.Types ( Data/Text/ICU/Types.hs, dist/build/Data/Text/ICU/Types.o, dist/build/Data/Text/ICU/Types.dyn_o )
cbits/text_icu.c: In function ‘__hs_unorm_quickCheck’:

cbits/text_icu.c:265:5: error:
     warning: ‘unorm_quickCheck’ is deprecated [-Wdeprecated-declarations]
      265 |     return unorm_quickCheck(source, sourcelength, mode, status);
          |     ^~~~~~
    |
265 |     return unorm_quickCheck(source, sourcelength, mode, status);
    |     ^
In file included from /usr/include/unicode/ucol.h:17,
                 from include/hs_text_icu.h:9,

                 from cbits/text_icu.c:1:0: error: 

/usr/include/unicode/unorm.h:244:1: error:
     note: declared here
      244 | unorm_quickCheck(const UChar *source, int32_t sourcelength,
          | ^~~~~~~~~~~~~~~~
    |
244 | unorm_quickCheck(const UChar *source, int32_t sourcelength,
    | ^
cbits/text_icu.c: In function ‘__hs_unorm_isNormalized’:

cbits/text_icu.c:272:5: error:
     warning: ‘unorm_isNormalized’ is deprecated [-Wdeprecated-declarations]
      272 |     return unorm_isNormalized(src, srcLength, mode, pErrorCode);
          |     ^~~~~~
    |
272 |     return unorm_isNormalized(src, srcLength, mode, pErrorCode);
    |     ^
In file included from /usr/include/unicode/ucol.h:17,
                 from include/hs_text_icu.h:9,

                 from cbits/text_icu.c:1:0: error: 

/usr/include/unicode/unorm.h:291:1: error:
     note: declared here
      291 | unorm_isNormalized(const UChar *src, int32_t srcLength,
          | ^~~~~~~~~~~~~~~~~~
    |
291 | unorm_isNormalized(const UChar *src, int32_t srcLength,
    | ^
cbits/text_icu.c: In function ‘__hs_unorm_normalize’:

cbits/text_icu.c:280:5: error:
     warning: ‘unorm_normalize’ is deprecated [-Wdeprecated-declarations]
      280 |     return unorm_normalize(source, sourceLength, mode, options, result,
          |     ^~~~~~
    |
280 |     return unorm_normalize(source, sourceLength, mode, options, result,
    |     ^
In file included from /usr/include/unicode/ucol.h:17,
                 from include/hs_text_icu.h:9,

                 from cbits/text_icu.c:1:0: error: 

/usr/include/unicode/unorm.h:218:1: error:
     note: declared here
      218 | unorm_normalize(const UChar *source, int32_t sourceLength,
          | ^~~~~~~~~~~~~~~
    |
218 | unorm_normalize(const UChar *source, int32_t sourceLength,
    | ^
cbits/text_icu.c: In function ‘__hs_u_strCompareIter’:

cbits/text_icu.c:308:43: error:
     error: ‘true’ undeclared (first use in this function)
      308 |     return u_strCompareIter(iter1, iter2, true);
          |                                           ^~~~
    |
308 |     return u_strCompareIter(iter1, iter2, true);
    |                                           ^

cbits/text_icu.c:308:43: error:
     note: each undeclared identifier is reported only once for each function it appears in
    |
308 |     return u_strCompareIter(iter1, iter2, true);
    |                                           ^
`x86_64-pc-linux-gnu-gcc' failed in phase `C Compiler'. (Exit code: 1)
 * ERROR: dev-haskell/text-icu-0.7.0.1::haskell failed (compile phase):
 *   setup build failed
 * 
 * Call stack:
 *     ebuild.sh, line  125:  Called src_compile
 *   environment, line 1948:  Called haskell-cabal_src_compile
 *   environment, line 1542:  Called cabal_src_compile
 *   environment, line  708:  Called cabal-build
 *   environment, line  466:  Called die
 * The specific snippet of code:
 *       ./setup "$@" || die "setup build failed"
 * 
 * If you need support, post the output of `emerge --info '=dev-haskell/text-icu-0.7.0.1::haskell'`,
 * the complete build log and the output of `emerge -pqv '=dev-haskell/text-icu-0.7.0.1::haskell'`.
 * The complete build log is located at '/var/tmp/portage/dev-haskell/text-icu-0.7.0.1/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/dev-haskell/text-icu-0.7.0.1/temp/environment'.
 * Working directory: '/var/tmp/portage/dev-haskell/text-icu-0.7.0.1/work/text-icu-0.7.0.1'
 * S: '/var/tmp/portage/dev-haskell/text-icu-0.7.0.1/work/text-icu-0.7.0.1'
```

Thx a lot :)

**Comment by trofi** [comment]
Are you using `=dev-libs/icu-67.1` by chance?

**Comment by hereGittyGitty** [comment]
> Are you using =dev-libs/icu-67.1 by chance?

Yes

`equery list dev-libs/icu`

```                                                                                                                                                                                    
[IP-] [  ] dev-libs/icu-67.1:0/67.1 
```

**Comment by trofi** [comment]
Can you try if https://github.com/gentoo-haskell/gentoo-haskell/commit/675b5d63904da7f9bc03a4519f869536256a0d10 fixes it for you?

**Comment by escalopoulet** [comment]
Hi.

I was having the same problem as hereGittyGitty. The fix given by trofi works fine for you.
Thank you trofi !
François.

**Comment by hereGittyGitty** [comment]
Hi,

I was able to check it out now. Yes, the fix in commit [675b5d6](https://github.com/gentoo-haskell/gentoo-haskell/commit/675b5d63904da7f9bc03a4519f869536256a0d10) did it. Thanks you trofi for the quick and nice handling. Thanks escalopoulet for letting me know.

Have a good one :)
